src/sentry/runner/commands/workstations.py

```diff
@@ -0,0 +1,576 @@
+    """
+
+    found = next((c for c in _configs(project) if c.config == config), None)
+    if found is None:
+        raise WorkstationsException(f"Could not find the {config} config in project {project}")
+
+    return found
+
+
+def _notify(text: str) -> None:
+    """
+    Prints simple status updates to the console in a highlighted style.
+    """
+
+    lines = text.splitlines()
+    if len(lines) == 0:
+        return
+
+    head = lines[0]
+    tail = lines[1:]
+    click.echo(click.style(f"➤  {head}", bold=True, italic=True, fg="cyan"))
+    for line in tail:
+        click.echo(click.style(f"   {line.strip()}", bold=True, fg="cyan"))
+
+
+def _get_workstation_info(name: str, conf: WorkstationConfig) -> WorkstationInfo | None:
+    """
+    Pull down the latest information for a given workstation.
+    """
+
+    result = _sync_gcloud_workstation_cmd(
+        [
+            "workstations",
+            "list",
+            "--filter",
+            f"NAME ~ '{name}$' AND CLUSTER ~ '{conf.cluster}'",
+            "--format",
+            "csv[no-heading](CONFIG,STATE)",
+        ],
+        scope=conf.project,
+        timeout=10,
+    )
+    lines = result.stdout.splitlines()
+    if len(lines) == 0:
+        return None
+    if len(lines) > 2:
+        raise WorkstationsException(
+            f"Encountered multiple workstations with the same name '{name}'"
+        )
+
+    (_, state) = lines[0].split(",")
+    return WorkstationInfo(
+        name=name,
+        state=WorkstationState.from_string(state),
+        config=conf,
+    )
+
+
+def _get_open_port() -> int:
+    """
+    Finds a random open port between 1024 and 65535.
+    """
+
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    s.bind(("", 0))
+    _, port = s.getsockname()
+    s.close()
+    return port
+
+
+@contextmanager
+def gcloud_manager(ctx: click.Context, project: str) -> Generator[None, None, None]:
+    """
+    Handles call(s) into the `gcloud` binary gracefully.
+    """
+
+    try:
+        _check_gcloud_setup(project)
+        yield
+        ctx.exit(0)
+    except WorkstationsException as e:
+        click.echo(message=str(e), err=True)
+        ctx.exit(1)
+    except subprocess.CalledProcessError:
+        # Assumes that piping the output from the `gcloud` command(s) to the shell is sufficient for
+        # error reporting to the user.
+        ctx.exit(2)
+    except subprocess.TimeoutExpired:
+        click.echo(message=ERR_TIMEOUT, err=True)
+        ctx.exit(3)
+
+
+@click.group()
+def workstations() -> None:
+    """
+    Create a bespoke Google Cloud Workstation instance.
+
+    NOTE: THIS IS CURRENTLY AN ALPHA STAGE COMMAND! If you are a Sentry dev, you'll need explicit
+    permissions from the OSPO team to spin up a workstation. Please reach out to us on
+    #discuss-self-hosted - we'd be happy to get you set up! :)
+
+    This command is primarily intended for Sentry employees working on self-hosted. The
```

Comment:
Perhaps it would be useful to link to https://github.com/getsentry/self-hosted/blob/master/workstation/README.md somewhere in the help section for the command?

Reply:
Done.